### PR TITLE
Make submodule inputs void in ExpandWhens

### LINKED ISSUE #705: latest FIRRTL generates incorrect code in rocket-chip
See https://github.com/freechipsproject/rocket-chip/issues/1160

### PR DESCRIPTION
**Pull request by jackkoenig**
Fixes #705 

The ports of submodules were not voided in ExpandWhens which means it does not error if they are not properly initialized. Worse still, since they don't have a default value in the netlist, the logic in ExpandWhens just picks the first connection and ignores all the rest.

I've added a test for each of the above cases.

### CODE REVIEW

File: src/main/scala/firrtl/passes/ExpandWhens.scala [hunk 1]
```diff
@@ -103,12 +103,16 @@ object ExpandWhens extends Pass {
...
                       (s: Statement): Statement = s match {
+        case stmt @ (_: DefNode | EmptyStmt) => stmt
         case w: DefWire =>
           netlist ++= (getFemaleRefs(w.name, w.tpe, BIGENDER) map (ref => we(ref) -> WVoid))
           w
         case w: DefMemory =>
           netlist ++= (getFemaleRefs(w.name, MemPortUtils.memType(w), MALE) map (ref => we(ref) -> WVoid))
           w
+        case w: WDefInstance =>
+          netlist ++= (getFemaleRefs(w.name, w.tpe, MALE).map(ref => we(ref) -> WVoid))
```
Comment (colinschmidt) [inline src/main/scala/firrtl/passes/ExpandWhens.scala]: `MALE` of `getFemaleRefs` here means to get all the inputs to the submodule?

Comment (colinschmidt) [reply src/main/scala/firrtl/passes/ExpandWhens.scala]: Clarified offline

Comment (jackkoenig) [reply src/main/scala/firrtl/passes/ExpandWhens.scala]: That's correct, similar to how it's used on the `DefMemory` above.

**Comment by jackkoenig** [comment]
Due to https://github.com/freechipsproject/chisel3/issues/746 this currently doesn't work with rocket-chip, I am trying to get this fixed tonight

**Pull request merged**